cmd/root.go

```diff
@@ -121,14 +124,21 @@ func init() {
 }
 
 func initConfig() {
-	if cfgFile != "" {
-		viper.SetConfigFile(cfgFile)
-		if err := viper.ReadInConfig(); err == nil {
-			log.Printf("Using config file: %s", viper.ConfigFileUsed())
-		} else {
-			log.Fatal(fmt.Sprintf("Couldn't read config file: %s", err.Error()))
-		}
-	} else {
+	if cfgFile == "" {
 		log.Warn("No config file passed with --config flag")
+		return
+	}
+
+	viper.SetConfigFile(cfgFile)
+	if err := viper.ReadInConfig(); err != nil {
+		log.Fatalf("Couldn't read config file: %s", err.Error())
+	}
+
+	log.Infof("Using config file: %s", viper.ConfigFileUsed())
+
+	if filepath.Ext(cfgFile) == ".env" {
```

Comment:
What if we want to load some params from a .toml and others from a .env? These files won't necessarily be the same, I think we should introduce a new CLI flag for specifying a ENV file path distinct from the TOML config file path. What do you think?

Reply:
I will update the code to use a different flag for CLI.